cmd/prysmctl/validator/withdraw.go

```diff
@@ -0,0 +1,150 @@
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+	"go.opencensus.io/trace"
+)
+
+func setWithdrawalAddresses(c *cli.Context) error {
+	ctx, span := trace.StartSpan(c.Context, "withdrawal.setWithdrawalAddress")
+	defer span.End()
+	au := aurora.NewAurora(true)
+	beaconNodeHost := c.String(BeaconHostFlag.Name)
+	if !c.IsSet(PathFlag.Name) {
+		return fmt.Errorf("no --%s flag value was provided", PathFlag.Name)
+	}
+	setWithdrawalAddressJsons, err := getWithdrawalMessagesFromPathFlag(c)
+	if err != nil {
+		return err
+	}
+	for _, request := range setWithdrawalAddressJsons {
+		fmt.Println("SETTING VALIDATOR INDEX " + au.Red(request.Message.ValidatorIndex).String() + " TO WITHDRAWAL ADDRESS " + au.Red(request.Message.ToExecutionAddress).String())
+	}
+	return callWithdrawalEndpoints(ctx, beaconNodeHost, setWithdrawalAddressJsons)
+}
+
+func getWithdrawalMessagesFromPathFlag(c *cli.Context) ([]*apimiddleware.SignedBLSToExecutionChangeJson, error) {
+	setWithdrawalAddressJsons := make([]*apimiddleware.SignedBLSToExecutionChangeJson, 0)
+	foundFilePaths, err := findWithdrawalFiles(c.String(PathFlag.Name))
+	if err != nil {
+		return setWithdrawalAddressJsons, errors.Wrap(err, "failed to find withdrawal files")
+	}
+	for _, foundFilePath := range foundFilePaths {
+		b, err := os.ReadFile(filepath.Clean(foundFilePath))
+		if err != nil {
+			return setWithdrawalAddressJsons, errors.Wrap(err, "failed to open file")
+		}
+		var to []*apimiddleware.SignedBLSToExecutionChangeJson
+		if err := json.Unmarshal(b, &to); err != nil {
+			log.Warnf("provided file: %s, is not a list of signed withdrawal messages", foundFilePath)
+			continue
+		}
+		setWithdrawalAddressJsons = append(setWithdrawalAddressJsons, to...)
+	}
+	if len(setWithdrawalAddressJsons) == 0 {
+		return setWithdrawalAddressJsons, errors.New("the list of signed requests is empty")
+	}
+	return setWithdrawalAddressJsons, nil
+}
+
+func callWithdrawalEndpoints(ctx context.Context, host string, request []*apimiddleware.SignedBLSToExecutionChangeJson) error {
+	client, err := beacon.NewClient(host)
+	if err != nil {
+		return err
+	}
+	if err := client.SubmitChangeBLStoExecution(ctx, request); err != nil {
+		return err
+	}
+	log.Infof("Successfully published messages to update %d withdrawal addresses.", len(request))
+	return checkIfWithdrawsAreInPool(ctx, client, request)
+}
+
+func checkIfWithdrawsAreInPool(ctx context.Context, client *beacon.Client, request []*apimiddleware.SignedBLSToExecutionChangeJson) error {
+	log.Info("verifying requested withdrawal messages known to node...")
+	poolResponse, err := client.GetBLStoExecutionChanges(ctx)
+	if err != nil {
+		return err
+	}
+	requestMap := make(map[string]string)
+	for _, w := range request {
+		requestMap[w.Message.ValidatorIndex] = w.Message.ToExecutionAddress
+	}
+	for _, resp := range poolResponse.Data {
+		value, found := requestMap[resp.Message.ValidatorIndex]
+		if found && value == resp.Message.ToExecutionAddress {
```

Comment:
I propose a slightly different implementation. Issue a warning for every missing change and issue an optional info log when all changes are in the pool. Having a list of continuous missing changes is more useful to me than intertwined success/failure logs.

Reply:
that's a good point I'll change to this then.